proto/viam/app/v1/app.proto

```diff
@@ -870,11 +883,14 @@ message ListFragmentsResponse {
 message GetFragmentRequest {
   string id = 1;
   string current_organization_id = 2;
+  optional string version = 3;
 }
 
 message GetFragmentResponse {
   Fragment fragment = 1;
-  FragmentUsage fragment_usage = 2;
+  repeated FragmentUsage usages = 2;
```

Comment:
i think this is a breaking change, but i think it should be okay if it's resolved before the list machines using fragment epic is launched

Reply:
I think the reason we initially did a single item is that we could return a mapping of version usage in the FragmentUsage message. I don't think the [Fragment Version Tech Spec](https://docs.google.com/document/d/1K71WS1a1t9alIf3JJ3fHQr9Nq-ltqPmTTxz4G11vaFY/edit?tab=t.0) prescribes the exact message but we are planning on storing it as a map based on the [Fragment Usage Stats Tech Spec](https://docs.google.com/document/d/1zw_bQl4BdU7V6XitxfkrG6ML0b4FtQgBDNX45_PtLUg/edit?tab=t.0#heading=h.kbgutx5vgkgq).

Reply:
oh hm, were you thinking something like this? for my understanding, is the primary advantage of this that it's closer to our DB storage schema?
```proto
message FragmentUsageStats {
  int32 organizations = 2;
  int32 machines = 3;
  int32 machines_in_current_org = 4;
}

message FragmentUsage {
  string fragment_id = 1;
  map<string, FragmentUsageStats> usage = 2;
}
```

Reply:
I think there are a few options. Mainly we have to decide what the right balance between returning overall metrics along with version level metrics. I believe we still need to return overall metrics for a few reasons, mainly that the Fragment list page will show overall metrics:
1. The way you have it in the PR code now, all metrics are versioned and the only way to get overall metrics (usage across all versions) returned is to put in a key like "all" which isn't necessarily clean.
2. What you proposed above but then we have a similar issue with needing to store "all".
3. Could do this which has fields for revisions and overall:

```
message FragmentUsageStats {
  int32 organizations = 2;
  int32 machines = 3;
  int32 machines_in_current_org = 4;
}

message FragmentUsage {
  string fragment_id = 1;
  FragmentUsageStats fragment_usage = 2;
  map<string, FragmentUsageStats> fragment_revision_usage = 3;
}
```

LMK what you think. We could also change `fragment_usage` to `overall_usage` to be more clear but the way I wrote it out is not a breaking change.

Reply:
ah, okay. i didn't realize that we also needed to show overall metrics. this makes sense!

Reply:
@michaellee1019 i updated to your suggestion. i renamed to `overall_usage` because i think it's a breaking change anyway to move the stats out of the original `FragmentUsage` message